internal/hammer/loadtest/workers.go

```diff
@@ -231,3 +275,187 @@ func (w *LogWriter) Kill() {
 		w.cancel()
 	}
 }
+
+// NewMMDVerifier creates a MMDVerifier.
+func NewMMDVerifier(tracker *client.LogStateTracker, mmdDuration time.Duration, errChan chan<- error, leafMMDChan <-chan LeafMMD) *MMDVerifier {
+	return &MMDVerifier{
+		tracker:     tracker,
+		mmdDuration: mmdDuration,
+		errChan:     errChan,
+		leafMMDChan: leafMMDChan,
+	}
+}
+
+// MMDVerifier verifies the signed timestamp against the MMD policy for newly
+// added entries by performing inclusion proof.
+type MMDVerifier struct {
+	tracker     *client.LogStateTracker
+	mmdDuration time.Duration
+	errChan     chan<- error
+	leafMMDChan <-chan LeafMMD
+	cancel      func()
+}
+
+// Run runs the MMD verifier. This should be called in a goroutine.
+func (v *MMDVerifier) Run(ctx context.Context) {
+	if v.cancel != nil {
+		panic("MMDVerifier was ran multiple times")
+	}
+	ctx, v.cancel = context.WithCancel(ctx)
+
+	var leafMMD *LeafMMD // The leaf we are currently processing, if any.
+	for {
+		if leafMMD == nil {
+			select {
+			case <-ctx.Done():
+				return
+			case mmd, ok := <-v.leafMMDChan:
+				if !ok {
+					return
+				}
+				leafMMD = &mmd
+			default:
+				// Nothing in the leafMMDChan, wait a bit and try again.
```

Comment:
The purpose of selecting from the two channels above is that you peacefully block until cancellation or availability of work. This default case causes this to be busier by running around the loop more often... why?

Reply:
Ah right, I've removed it.